File: algorithms/Recursion/DP/recDC.py
def recDC(coinValueList, change, knownResult):
    minCoins = change
    if change in coinValueList:
        return 1
    elif knownResult[change] > 0:
        return knownResult[change]
    else:
        for i in [c for c in coinValueList if c <= change]:
            numCoins = 1 + recDC(coinValueList, change - i, knownResult)

            if numCoins < minCoins:
                minCoins = numCoins
                knownResult[change] = minCoins
    return minCoins

print(recDC([1, 5, 10, 25], 63, [0]*64))

File: algorithms/Recursion/DP/recDC.js
function recDC(coinValueList, change, knownResult) {
    var minCoin = change
    if (coinValueList.indexOf(change) != -1) {
        return 1;
    } else if(knownResult[change] > 0) {
        return knownResult[change]
    } else {
        coinValueList.filter(function(value, index) {
            if(value <= change) {
                return value
            }    
        })
        .forEach(function(value, index) {
            numConis = 1 + recDC(coinValueList, change - value, knownResult)
            if (numConis < minCoin) {
                minCoin = numConis
                knownResult[change] = numConis
            }
        })
    }
    return minCoin
}

console.log(recDC([1, 5, 10, 25], 63, new Array()))